k8s/upgrade/src/preflight_validations.rs

```diff
@@ -168,15 +168,29 @@ pub async fn is_rebuild_in_progress(client: &upgrade_lib::RestClient) -> error::
 /// Struct to deserialize the unsupported version yaml.
 #[derive(Debug, Serialize, Deserialize)]
 struct UnsupportedVersions {
-    unsupported_versions: Vec<String>,
+    unsupported_versions: Vec<Version>,
 }
 
-pub async fn upgrade_path_validation(namespace: &str) -> error::Result<()> {
-    let path = std::env::current_dir()
-        .context(error::GetCurrentDirectory)?
-        .join(UNSUPPORTED_VERSION_FILE);
+impl UnsupportedVersions {
+    fn contains(&self, v: &Version) -> bool {
+        self.unsupported_versions.contains(v)
+    }
+}
+
+impl TryFrom<&[u8]> for UnsupportedVersions {
+    type Error = crate::error::Error;
 
-    let unsupported_versions = get_unsupported_versions(path)?;
+    /// Returns an UnsupportedVersions object.
+    fn try_from(bytes: &[u8]) -> error::Result<Self> {
+        let mut path = PathBuf::new();
+        path.push(UNSUPPORTED_VERSION_FILE);
```

Comment:
let path = Path::new(UNSUPPORTED_VERSION_FILE).to_path_buf();

Reply:
Actually, this is a bit misleading here, because bytes is just an array of bytes, I think the context should be added by the calling function!

Reply:
IMHO, `serde_yaml::from_reader` is reading the bytes and any error occurs while parsing should be thrown. Since the parsing is of the unsupported_version.yaml file the error is thrown about the same.

Reply:
it would be thrown, but it would be the original error from serde_yaml... you need only attach the error you want with the path at the calling side. 
Let say we try to load a different file into UnsupportedVersions, the error message is going to report the wrong path!